queenbee/recipe/dag.py

```diff
@@ -310,7 +311,7 @@ class DAGTaskOutputArtifact(BaseModel):
     )
 
     path: str = Field(
-        ...,
+        None,
```

Comment:
When your task is a `function` you can set a path. However when it is another `dag` it becomes very difficult to pass down location information for specific artifacts as most workflows will have their own internal logic and artifact placement. In that case I would say it's better to leave the `path` as an optional thing (that has not effect when you run a `dag` task). We can add a check for the template type and throw a warning if you want.

If the task is a function and no path is given I guess we can just save the `artifact` to some random/semi-deterministic folder?

Reply:
I can't fully comprehend the case that you mentioned however that is most likely is because I haven't played with using dags as a task. I think since this is an output there should be a path regardless and the output should be copied somewhere.

I trust your judgment here and I think I will see it when I start working with more complex recipes. 😀 Meanwhile can we add a warning as you mentioned? I can see this to be an issue if we do not enforce the path for functions since the validation will pass but the execution can fail.

Reply:
Ah! Maybe worth making sure we're clear on the use of the `path` variable here.

> For a Task, the output artifact `path` corresponds to the path where the artifact is persisted within the `run folder`. 

So Operator Functions still have to provide a path where they expect to find the artifact in the context of the `function folder`. (ie: for Argo this will be the path inside the container).

We can chat about it in person if you need it. However just trusting my judgement is also fine (:fire: :fire: :fire:)

Reply:
Do you have a sample recipe where the path is not required? That should clarify this for me. 😀

You don't have to wait for me to figure this out though to merge this PR in.

Reply:
Here's an example of a `parametric` daylight factor run:

```yaml
name: main
inputs:
  parameters:
  - name: model-names
    descriptions: a list of model folder names
  artifacts:
  - name: models-folder
    description: A folder full of honeybee models

tasks:

- name: daylight-factor
  template: daylight-factor
  loop:
    from:
      type: inputs
      variable: model-names
    control:
      key: '{{item.name}}'
  sub_folder: daylight-factor/{{item.name}}
  arguments:
    artifacts:
    - name: model
      from:
        type: inputs
        variable: models-folder
      subpath: '{{item}}/model'
    - name: input-grid
      from:
        type: inputs
        variable: models-folder
      subpath: '{{item}}/grid'
  outputs:
    artifacts:
    - name: result-grid

outputs:
  artifacts:
  - name: daylight-factor-grids
    from:
      type: tasks
      name: daylight-factor
      variable: result-grid
```